numpy/core/src/umath/ufunc_object.c

```diff
@@ -4169,20 +4172,31 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
         return (PyObject *)ret;
     }
 
-    if (Py_TYPE(op) != Py_TYPE(ret)) {
-        res = PyObject_CallMethod(op, "__array_wrap__", "O", ret);
-        if (res == NULL) {
-            PyErr_Clear();
+    else {
+        PyObject *wrap;
+        _find_array_wrap(args, kwds, &wrap, 1, 1);
```

Comment:
Turns out this doesn't work, as it interprets the `axis` argument as an `out` argument